src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Persistence/Query/Completeness/GetNonRequiredAttributesMasksQuery.php

```diff
@@ -28,28 +28,46 @@ public function __construct(Connection $connection)
     public function fromFamilyCodes(array $familyCodes): array
     {
         $sql = <<<SQL
+WITH
+table_column_exists AS (
+    SELECT EXISTS (
+       SELECT * FROM information_schema.tables
+       WHERE table_name = 'pim_catalog_table_column'
+    )
+)
 SELECT
     family.code AS family_code,
     channel_code,
     locale_code,
     JSON_ARRAYAGG(
         CONCAT(
-            IF(
-                attribute.attribute_type = 'pim_catalog_price_collection',
-                CONCAT(
-                    attribute.code,
-                    '-',
-                    (
-                        SELECT GROUP_CONCAT(currency.code ORDER BY currency.code SEPARATOR '-')
-                        FROM pim_catalog_channel channel
-                            JOIN pim_catalog_channel_currency pccc ON channel.id = pccc.channel_id
-                            JOIN pim_catalog_currency currency ON pccc.currency_id = currency.id
-                        WHERE channel.code  = channel_code
-                        GROUP BY channel.id
-                    )
-                    ),
-                attribute.code
-                ),
+            CASE
+                WHEN attribute.attribute_type = 'pim_catalog_price_collection' 
+                    THEN CONCAT(
+                            attribute.code,
+                            '-',
+                            (
+                                SELECT GROUP_CONCAT(currency.code ORDER BY currency.code SEPARATOR '-')
+                                FROM pim_catalog_channel channel
+                                JOIN pim_catalog_channel_currency pccc ON channel.id = pccc.channel_id
+                                JOIN pim_catalog_currency currency ON pccc.currency_id = currency.id
+                                WHERE channel.code  = channel_code
+                                GROUP BY channel.id
+                            )
+                        )
+                WHEN attribute.attribute_type = 'pim_catalog_table' 
```

Comment:
Should we test the `table_column_exists` in this condition?
```suggestion
                WHEN table_column_exists = '1' AND attribute.attribute_type = 'pim_catalog_table'
```